### PR TITLE
fix: update usage doc for all oras commands

### LINKED ISSUE #593: unify styling for all command help documents
# Usage Notation
Reference: https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/command-line-syntax-key
- Text without brackets or braces: Items you must type as shown.
- `<Text inside angle brackets>`: Placeholder for which you must supply a value.
- `[Text inside square brackets]`: Optional items.
- `{Text inside braces}`: Set of required items. You must choose one.
- Vertical bar `( \| )`: Separator for mutually exclusive items. You must choose one.
- Ellipsis `(…)`: Items that can be repeated and used multiple times.

# Styling
- Flags should be put before arguments

### LINKED ISSUE #602: Improve `oras attach` help doc
Command:
```
oras attach -h
```
Output:
image

Users can't tell that `--artifact-type` is a required field from the current help doc.
Besides, the usage format `name<:tag|@digest>` is not consistent with other commands like `manifest fetch`.
image

Some improvement can be like:
1. Add `--artifact-type` into the usage doc:
```diff
- oras attach <name:tag|name@digest> [file[:type]...] [flags]
+ oras attach <name:tag|name@digest> [file[:type]...] --artifact-type <artifactType> [flags]
```
2. Call out that `--artifact-type` is a required filed in the flag description:
```diff
- artifact type
+ artifact type (required)
```

### PR DESCRIPTION
**Pull request by qweeah**
Resolves #593, resolves #602

Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: cmd/oras/attach.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type attachOptions struct {
 func attachCmd() *cobra.Command {
 	var opts attachOptions
 	cmd := &cobra.Command{
-		Use:   "attach name<:tag|@digest> [file[:type]...]",
+		Use:   "attach [flags] <name>{:<tag>|@<digest>} [<file>[:<type>]...]",
```
Comment (shizhMSFT) [inline cmd/oras/attach.go:47]:
```suggestion
		Use:   "attach [flags] <name>{:tag|@digest} [file[:type]] ...",
```

Comment (qweeah) [reply cmd/oras/attach.go:47]: According to the syntax [guideline](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/command-line-syntax-key), we should not, e.g. https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/cd

File: cmd/oras/attach.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ type attachOptions struct {
 func attachCmd() *cobra.Command {
 	var opts attachOptions
 	cmd := &cobra.Command{
-		Use:   "attach name<:tag|@digest> [file[:type]...]",
+		Use:   "attach [flags] <name>{:<tag>|@<digest>} {<file>[:<type>]}[...]",
```
Comment (shizhMSFT) [inline cmd/oras/attach.go:47]: Can we close #603 and merge the change here? Also, update the description to resolve #602?
```suggestion
		Use:   "attach [flags] --artifact-type=<type> <name>{:<tag>|@<digest>} [<file>[:<type>]] [...]",
```

File: cmd/oras/blob/fetch.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ type fetchBlobOptions struct {
 func fetchCmd() *cobra.Command {
 	var opts fetchBlobOptions
 	cmd := &cobra.Command{
-		Use:   "fetch [flags] <name@digest>",
+		Use:   "fetch [flags] {--output <file>|--descriptor} <name>@<digest>",
```
Comment (shizhMSFT) [inline cmd/oras/blob/fetch.go:45]: Should we add spaces around `|`? I'm just wondering.
```suggestion
		Use:   "fetch [flags] {--output <file> | --descriptor} <name>@<digest>",
```

File: cmd/oras/blob/push.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ type pushBlobOptions struct {
 func pushCmd() *cobra.Command {
 	var opts pushBlobOptions
 	cmd := &cobra.Command{
-		Use:   "push [flags] name[@digest] file",
+		Use:   "push [flags] <name>[@digest] <file>",
```
Comment (shizhMSFT) [inline cmd/oras/blob/push.go:45]:
```suggestion
		Use:   "push [flags] <name>[@<digest>] <file>",
```

Comment (shizhMSFT) [inline cmd/oras/blob/push.go:45]: I just found a UX inconsistency here.

We should remove the `--media-type` option. Instead, we should
```suggestion
		Use:   "push [flags] <name>[@digest] <file>[:<type>]",
```

/cc @qweeah @lizMSFT @yuehaoliang-microsoft @FeynmanZhou

File: cmd/oras/manifest/push.go [hunk 5]
```diff
@@ -43,7 +43,7 @@ type pushOptions struct {
 func pushCmd() *cobra.Command {
 	var opts pushOptions
 	cmd := &cobra.Command{
-		Use:   "push [flags] name[:tag|@digest] file",
+		Use:   "push [flags] <name>[:<tag>|@<digest>] <file>",
```
Comment (shizhMSFT) [inline cmd/oras/manifest/push.go:46]: I just found a UX inconsistency here.

We should remove the `--media-type` option. Instead, we should
```suggestion
		Use:   "push [flags] <name>[:<tag>|@<digest>] <file>[:<type>]",
```
/cc @qweeah @lizMSFT @yuehaoliang-microsoft @FeynmanZhou

File: cmd/oras/push.go [hunk 6]
```diff
@@ -46,7 +46,7 @@ type pushOptions struct {
 func pushCmd() *cobra.Command {
 	var opts pushOptions
 	cmd := &cobra.Command{
-		Use:   "push name[:tag|@digest] file[:type] [file...]",
+		Use:   "push [flags] <name>[:<tag>|@<digest>] <file>[:type] [...]",
```
Comment (shizhMSFT) [inline cmd/oras/push.go:49]:
```suggestion
		Use:   "push [flags] <name>[:<tag>|@<digest>] <file>[:<type>] [...]",
```

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: cmd/oras/blob/fetch.go (hunk 3 above)
Comment (shizhMSFT) [reply cmd/oras/blob/fetch.go:45]: Feel free to defend it.

Comment (qweeah) [reply cmd/oras/blob/fetch.go:45]: We should. The reason is simple: it looks better.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**